servers/src/main/java/tachyon/master/block/BlockMaster.java

```diff
@@ -561,6 +573,17 @@ private void processWorkerRemovedBlocks(MasterWorkerInfo workerInfo,
   }
 
   /**
+   * Called by the heartbeat thread whenever a worker is lost.
+   * @param latest the latest {@link MasterWorkerInfo} available at the time of worker loss
+   */
+  private void processLostWorker(MasterWorkerInfo latest) {
+    synchronized (mBlocks) {
```

Comment:
I think there could be a deadlock with this. The heartbeat thread locks mWorkers first, then calls processLostWorker() which locks mBlocks.

However, I think when both need to be locked, mBlocks needs to be locked first.